### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by AinaArd**

### CODE REVIEW

File: src/test/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+spring.datasource.driver-class-name=org.h2.Driver
+spring.datasource.url=jdbc:h2:mem:db;DB_CLOSE_DELAY=-1
+spring.datasource.username=sa
+spring.datasource.password=sa
+spring.jpa.hibernate.ddl-auto=update
+
+token.expired=86400
```
Comment (Dimanaux) [inline src/test/resources/application.properties]: Do you ever use this setting?

File: src/test/java/ru/itis/services/WishListServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private WishListService wishListService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    public void addWishListWorksOk() {
+        wishListService.addNewWishList(wishListDto.getTitle(), token);
+        Optional<WishList> found = wishListService.findWishListByTitle(wishListDto.getTitle());
+        found.ifPresent(list -> assertEquals(wishListDto.getTitle(), list.getTitle()));
```
Comment (Dimanaux) [inline src/test/java/ru/itis/services/WishListServiceTest.java]: It tests nothing. If wishListService fails to save, your test will pass. This test is bad.

File: src/main/java/ru/itis/services/WishListService.java [hunk 3]
```diff
@@ -36,4 +37,13 @@ public boolean removeByTitle(String title, String token) {
...
         return false;
     }
+
+    public Optional<WishList> findWishListByTitle(String title) {
+        return wishListRepository.findByTitle(title);
+    }
+
+    public List<WishList> findAllByAuthor(String token) {
+        Optional<User> user = userService.findUserByToken(token);
+        return wishListRepository.findAllByAuthor(user.get());
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: Why not author.wishLists()?

File: src/main/java/ru/itis/services/WishListService.java [hunk 4]
```diff
@@ -36,4 +37,13 @@ public boolean removeByTitle(String title, String token) {
         wishListCandidate.ifPresent(wishList -> wishListRepository.delete(wishList));
         return false;
     }
+
+    public Optional<WishList> findWishListByTitle(String title) {
+        return wishListRepository.findByTitle(title);
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/WishListService.java]: Why do you need this class if it only delegates?

File: src/main/java/ru/itis/services/UserService.java [hunk 5]
```diff
@@ -67,11 +67,11 @@ public TokenDto login(UserDto userDto) {
 
     public Optional<User> findUserByToken(String token) {
         String userLogin = tokensRepository.findUsernameByValue(token);
-        return usersRepository.findByLogin(userLogin);
+        return userRepository.findByLogin(userLogin);
     }
 
     public List<UserDto> getUsersByLogin(String login) {
-        return usersRepository.findAllByLoginIsContaining(login);
+        return userRepository.findAllByLoginIsContaining(login);
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/UserService.java]: Why do you need this class if it only delegates?

File: src/main/java/ru/itis/controllers/ProfileController.java [hunk 6]
```diff
@@ -30,32 +28,28 @@ public ProfileController(WishListService wishListService, UserService userServic
...
 
     @CrossOrigin
     @DeleteMapping("/profile")
     @PreAuthorize("isAuthenticated()")
     @ApiOperation("Delete a wish list")
-    public ResponseEntity<?> deleteWishList(@RequestParam String title, @RequestHeader("Authorization") String token) {
+    public ResponseDto deleteWishList(@RequestParam String title, @RequestHeader("Authorization") String token) {
         if (!wishListService.removeByTitle(title, token)) {
-            return new ResponseEntity<>((HttpStatus.NOT_FOUND));
+            return new ResponseDto(HttpStatus.OK, "Invalid wish list title");
```
Comment (Dimanaux) [inline src/main/java/ru/itis/controllers/ProfileController.java]: Use http response status instead. https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success

File: pom.xml [hunk 7]
```diff
@@ -65,6 +65,22 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <version>1.4.197</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <version>2.1.3.RELEASE</version>
```
Comment (Dimanaux) [inline pom.xml]: `<scope>test</scope>`

File: pom.xml [hunk 8]
```diff
@@ -65,6 +65,22 @@
...
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <version>1.4.197</version>
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply pom.xml]: fixed

File: src/test/java/ru/itis/services/ItemServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ItemServiceTest {
+    private static ItemDto itemDto = new ItemDto("pencil", 10);
+
+    @Autowired
+    private ItemService itemService;
+
+    @Test
+    public void addItemWorksOk() {
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply src/test/java/ru/itis/services/ItemServiceTest.java]: what's better?

File: src/test/java/ru/itis/services/ItemServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        itemService.addNewItem(itemDto, 1L);
+        Optional<Item> found = itemService.findItemByName(itemDto.getName());
+        found.ifPresent(item -> assertEquals(itemDto.getName(), item.getName()));
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply src/test/java/ru/itis/services/ItemServiceTest.java]: added

File: src/test/java/ru/itis/services/UserServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+@SpringBootTest
+@Log4j2
+public class UserServiceTest {
+    private static UserDto userDto = new UserDto("test", "test1");
+
+    @Autowired
+    private UserService userService;
+
+    @Test
+    public void addUserWorksCorrect() {
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply src/test/java/ru/itis/services/UserServiceTest.java]: what's better?

File: src/test/resources/application.properties (hunk 1 above)
Comment (AinaArd) [reply src/test/resources/application.properties]: deleted

File: src/main/java/ru/itis/services/WishListService.java (hunk 3 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed

File: src/main/java/ru/itis/controllers/ProfileController.java (hunk 6 above)
Comment (AinaArd) [reply src/main/java/ru/itis/controllers/ProfileController.java]: why?

File: src/test/java/ru/itis/services/ItemServiceTest.java [hunk 12]
```diff
@@ -21,9 +21,10 @@
     private ItemService itemService;
 
     @Test
-    public void addItemWorksOk() {
+    public void addItemWorksGood() {
```
Comment (Dimanaux) [inline src/test/java/ru/itis/services/ItemServiceTest.java]: Still bad name. It is too abstract.

File: src/test/java/ru/itis/services/UserServiceTest.java [hunk 13]
```diff
@@ -25,13 +25,13 @@
     private UserService userService;
 
     @Test
-    public void addUserWorksCorrect() {
+    public void addUserWorksGood() {
         Optional<User> found = userService.findUserByLogin(userDto.getLogin());
         found.ifPresent(user -> assertEquals(user.getLogin(), userDto.getLogin()));
```
Comment (Dimanaux) [inline src/test/java/ru/itis/services/UserServiceTest.java]: Why only if present?

File: src/test/java/ru/itis/services/UserServiceTest.java [hunk 14]
```diff
@@ -25,13 +25,13 @@
     private UserService userService;
 
     @Test
-    public void addUserWorksCorrect() {
+    public void addUserWorksGood() {
```
Comment (Dimanaux) [inline src/test/java/ru/itis/services/UserServiceTest.java]: A bad name.

File: src/main/java/ru/itis/services/WishListService.java [hunk 15]
```diff
@@ -36,4 +37,13 @@ public boolean removeByTitle(String title, String token) {
...
+    }
+
+    public List<WishList> findAllByAuthor(String token) {
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: fixed the name of method, but how to fix the architecture?

File: src/test/java/ru/itis/services/WishListServiceTest.java (hunk 2 above)
Comment (AinaArd) [reply src/test/java/ru/itis/services/WishListServiceTest.java]: Actually, project doesn't have a lot of business logic, so test currently are not so important

File: src/main/java/ru/itis/services/UserService.java (hunk 5 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/UserService.java]: some actions can not be made in controller->need of service

File: src/main/java/ru/itis/services/WishListService.java (hunk 4 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/WishListService.java]: some actions can not be made in controller->need of service

**Pull request merged**